src/routes/api/collections/post.js

```diff
@@ -1,16 +1,68 @@
 const logger = require("../../../logger");
+const collection = require("../../../model/collection");
 const { createSuccessResponse } = require("../../../response");
 
 /**
  * Create a collection
  */
-
-// eslint-disable-next-line no-unused-vars
 module.exports = async (req, res, next) => {
   logger.debug(
     { user: req.user, body: req.body },
     `received request: POST /v1/collections`,
   );
 
-  return res.status(418).json(createSuccessResponse());
+  // Validate request format
+  try {
+    collection.schema.parse(req.body);
+  } catch (error) {
+    logger.warn({ error }, "invalid collection data");
+    const httpError = new Error("Invalid collection data");
+    httpError.status = 400;
+    return next(httpError);
+  }
+
+  let newCollection;
+
+  // Save collection
+  try {
+    newCollection = await collection.save(req.user, req.body);
+  } catch (error) {
+    return next(error);
+  }
+
+  logger.info(
+    { collectionId: newCollection.id, userId: req.user },
+    `user created new collection: "${newCollection.name}"`,
+  );
+
+  logger.debug(
+    { collection: newCollection },
+    `created new collection with id ${newCollection.id}`,
+  );
+
+  let locationURL;
+  // Create location URL
+  try {
+    locationURL = new URL(
+      `/v1/collections/${newCollection.id}`,
+      process.env?.API_URL || `https://${req.headers?.host}`,
+    );
+  } catch (error) {
+    logger.error(
+      { error, collection: newCollection },
+      "error constructing location URL",
+    );
+    return next(error);
+  }
```

Comment:
this doesn't need to be in a try/catch

```suggestion
  const locationURL = new URL(
    `/v1/collections/${newCollection.id}`,
    process.env?.API_URL || `https://${req.headers?.host}`,
  );
```